commands/Tools/Step 3 - Modify schematisation/control_structures.py

```diff
@@ -0,0 +1,171 @@
+from ThreeDiToolbox.threedi_schema_edits.controlled_structures import \
+    TABLE_CONTROL
+from ThreeDiToolbox.threedi_schema_edits.controlled_structures import \
+    ControlledStructures
+from ThreeDiToolbox.utils.threedi_database import get_databases
+from ThreeDiToolbox.utils.threedi_database import get_database_properties
+from ThreeDiToolbox.utils.user_messages import messagebar_message
+from ThreeDiToolbox.views.control_structures_dockwidget import ControlStructuresDockWidget  # noqa
+
+log = logging.getLogger(__name__)
+
+
+class CustomCommand(CustomCommandBase):
+    """
+    command that will load and start an edit session for the connected
+    point layer and verify the data added to that layer
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+        self.iface = kwargs.get('iface')
+        self.dockwidget_controlled_structures = None
+        self.control_structure = None
+
+        self.databases = get_databases()
+        # Remove 'selected' spatialite and postgresdatabases
+        # from self.databases to prevent confusion about which database
+        # is meant by it
+        if 'spatialite: ' in self.databases:
+            self.databases.pop('spatialite: ', None)
+        if 'postgres: None' in self.databases:
+            self.databases.pop('postgres: None', None)
+
+    def run(self):
+        """Run the controlled structures dockwidget."""
+        self.show_gui()
+        self.dockwidget_controlled_structures.combobox_input_model\
+            .activated.connect(self.update_dockwidget_ids)
+        self.dockwidget_controlled_structures.pushbutton_input_save\
+            .clicked.connect(self.run_it)
+
+    def run_it(self):
+        """Save the control to the spatialite or POSTGRES database."""
+        db_key = self.dockwidget_controlled_structures.combobox_input_model\
+            .currentText()  # name of database
+        db = get_database_properties(db_key)
+        control_structure = ControlledStructures(
+            flavor=db["db_entry"]['db_type'])
+        control_structure.start_sqalchemy_engine(db["db_settings"])
+        # The control_type is the type of control that will be saved.
+        # In the future, this type can be read from a combobox.
+        # Future options will be table control, pid control,
+        # memory control, delta control and timed control.
+        control_type = TABLE_CONTROL
+        if control_type == TABLE_CONTROL:
+            table_control = {}
+            measure_value = self.dockwidget_controlled_structures\
+                .tablewidget_input_rule_table_control.item(0, 0).text()
+            action_value = self.dockwidget_controlled_structures\
+                .tablewidget_input_rule_table_control.item(0, 1).text()
+            list_of_values = [measure_value, action_value]
+            table_control["action_table"] = control_structure\
+                .create_action_table(list_of_values)
```

Comment:
see my comment further below, why not:

``table_control["action_table"] = '{0};{1}'.format(measure_value, action_value)``